src/tiled/mapview.cpp

```diff
@@ -311,6 +311,46 @@ void MapView::mouseMoveEvent(QMouseEvent *event)
         return;
     }
 
+    if (event->buttons() & Qt::LeftButton) {
+        auto *hBar = static_cast<FlexibleScrollBar*>(horizontalScrollBar());
+        auto *vBar = static_cast<FlexibleScrollBar*>(verticalScrollBar());
+
+        QPoint pos = event->pos();
+        QRectF mRect = viewport()->rect();
+
+        int horizontalValue = 0,  verticalValue = 0;
+
+        if ((pos.x() + 5) > mRect.width()) {
+            horizontalValue = hBar->value() + 5;
+            cursor().setPos(mLastMousePos - *(new QPoint(5, 0)));
```

Comment:
I don't think it's a good idea to change the cursor position. I'm sure there's a better solution.